### PR TITLE
Add zookeeper driver

### PR DESCRIPTION
**Pull request by yuxuan-ms**

### CODE REVIEW

File: testplan/testing/multitest/driver/zookeeper.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    Configuration object for
+    :py:class:`~testplan.testing.multitest.driver.zookeeper.ZookeeperStandalone` resource.
+    """
+
+    @classmethod
+    def get_options(cls):
+        return {
+            ConfigOption("template"): str,
+            ConfigOption("binary"): str,
+            ConfigOption("port"): int,
```
Comment (Pyifan) [inline testplan/testing/multitest/driver/zookeeper.py:32]: What happens if user provide 0 port?

File: testplan/testing/multitest/driver/zookeeper.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        self.zklog_path = os.path.join(self.runpath, "zklog")
+        self.etc_path = os.path.join(self.runpath, "etc")
+        for directory in (self.zkdata_path, self.zklog_path, self.etc_path):
+            if self.cfg.path_cleanup is False:
+                makedirs(directory)
+            else:
+                makeemptydirs(directory)
+        self.generate_config()
+
+    def generate_config(self):
```
Comment (Pyifan) [inline testplan/testing/multitest/driver/zookeeper.py:83]: I think we can just use Driver._install_files() for this.

File: testplan/testing/multitest/driver/zookeeper.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            **options
+        )
+        self.port = port
+        self.config = None
+        self.zkdata_path = None
+        self.zklog_path = None
+        self.etc_path = None
+        self.std = None
+
+    def pre_start(self):
```
Comment (Pyifan) [inline testplan/testing/multitest/driver/zookeeper.py:71]: Let's add doc string for public APIs.

File: examples/DB/Zookeeper/test_plan.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python
+"""
+Demostrates Zookeeper driver usage from within the testcases.
+"""
+
+import os
+import sys
+
+try:
```
Comment (Pyifan) [inline examples/DB/Zookeeper/test_plan.py:9]: As we discussed, let's remove this try-except block (maybe add internal dependencies first?)

File: testplan/testing/multitest/driver/zookeeper.py (hunk 1 above)
Comment (yuxuan-ms) [reply testplan/testing/multitest/driver/zookeeper.py:33]: zookeeper doesn't support 0 port

File: testplan/testing/multitest/driver/zookeeper.py [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+            if self.cfg.path_cleanup is False:
+                makedirs(directory)
+            else:
+                makeemptydirs(directory)
+        self.config = os.path.join(self.runpath, "etc", "zookeeper.cfg")
+        if self.port == 0:
+            raise RuntimeError("Zookeeper doesn't support random port")
+        instantiate(self.cfg.template, self.context_input(), self.config)
+
+    def run_command(self, command):
```
Comment (Pyifan) [inline testplan/testing/multitest/driver/zookeeper.py:92]: oss/testplan/common/utils/process.py has an execute_cmd function. Maybe we can use it?

File: examples/DB/Zookeeper/test_plan.py [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    stdout_style=OUTPUT_STYLE,
+    pdf_style=OUTPUT_STYLE,
+    pdf_path="report.pdf",
+)
+def main(plan):
+    """
+    Testplan decorated main function to add and execute MultiTests.
+
+    :return: Testplan result object.
+    :rtype:  ``testplan.base.TestplanResult``
```
Comment (raoyitao) [inline examples/DB/Zookeeper/test_plan.py:74]: use ~ format that can link to the class

**Comment by yuxuan-ms** [comment]
I also updated setup.py and requirements-basic.txt. Because the newest version of valiators doesn't support Python2.

**Pull request merged**